llvm/lib/Target/RISCV/RISCVInstrInfo.td

```diff
@@ -1400,8 +1397,19 @@ def PseudoAddTPRel : Pseudo<(outs GPR:$rd),
 
 /// FrameIndex calculations
 
-def : Pat<(FrameAddrRegImm (iPTR GPR:$rs1), simm12:$imm12),
-          (ADDI GPR:$rs1, simm12:$imm12)>;
+// Converts frameindex -> tframeindex.
+def to_tframeindex : SDNodeXForm<frameindex, [{
+  return CurDAG->getTargetFrameIndex(N->getIndex(), N->getValueType(0));
+}]>;
+
+def : GICustomOperandRenderer<"renderFrameIndex">,
+      GISDNodeXFormEquiv<to_tframeindex>;
+
+def : Pat<(frameindex:$fi), (ADDI (iPTR (to_tframeindex $fi)), 0)>;
+
+def : Pat<(add_like frameindex:$fi, simm12:$offset),
```

Comment:
I think add_like might be less capable than `SelectionDAG::isBaseWithConstantOffset`, but if it shows up its probably fixable.

Reply:
`isBaseWithConstantOffset` additionally handles `(xor x, INT_MIN)`, that's all the difference I noticed.

Reply:
Given you think there is a difference, please don't mark this as "NFCI" in the message.

Reply:
I removed NFCI now